src/Raven.Client/Documents/Operations/CountersOperationExecutor.cs

```diff
@@ -81,14 +81,14 @@ public Task<CountersDetail> IncrementAsync(string docId, string name, long delta
             return details.Counters[0].TotalValue;
         }
 
-        public Dictionary<string, long> Get(string docId, IEnumerable<string> counters)
+        public Dictionary<string, long?> Get(string docId, IEnumerable<string> counters)
```

Comment:
Do we need CountersOperationExecutor at all? All now is a part of the session and we do not have equivalents for the rest. If the answer is 'yes' then we are missing there `SessionInfo sessionInfo = null` argument in each of those methods.

Reply:
Also all are OperationExecutors are based on IOperation or IServerOperation concept, but this one isnt.

Reply:
The one thing that we should be concerned here is the fact that:
* We need a lower level than the session
* Counters have different load balancing than other stuff, we likely will want to configure them separately.

Reply:
We can always use `store.Operations`.
If we want to have something different then I suggest renaming this to `DocumentStoreCounters` instead.

Reply:
@aviv86 Currently we _don't_ have a different behavior for load balancing on counters, right?

Reply:
right, we don't have a different behavior for load balancing on counters yet

Reply:
removed store.Counters
will recreate this in 4.2 with new policy
http://issues.hibernatingrhinos.com/issue/RavenDB-11348